mono/metadata/handle.h

```diff
@@ -196,10 +224,22 @@ Icall macros
 	mono_stack_mark_record_size (__info, &__mark, __FUNCTION__);	\
 	mono_stack_mark_pop (__info, &__mark);
 
+#if MONO_TYPE_SAFE_HANDLES
+
+// Structs cannot be cast to structs.
+// Therefore, cast the function pointer to change its return type.
+#define CLEAR_ICALL_FRAME_VALUE(type, handle)				\
+	((mono_stack_mark_record_size (__info, &__mark, __FUNCTION__)),	\
```

Comment:
Use `do { } while (0)` instead of a LISP style abuse of paranthesis.

Reply:
These have different meaning. Parens around expressions. do/while for statements. Use statements only where required. Prefer expressions for composability.
In at least one place I believe the composibility was required, i.e. because the type could not flow through a void*.

Reply:
Parens are not LISP style abuse. They are C style abuse. They are required to keep macro parameters separate from their surroundings. C++ fixes this with inline functions/templates.

Sometimes, alternatively, you can use temporaries, but this has drawbacks.

Once you have  temporaries, you lose the composibility of expressions, and you need to be able to state the types of the temporaries, which is sometimes difficult. There might be other problems. I never really thought deeply about it, or maybe long ago forgot details of the lesson, because everyone (except perhaps mono) just prefers expressions within macros, accepts that macros are poor, and therefore prefers neither.

Reply:
In particular, CLEAR_ICALL_FRAME_VALUE needs to be an expression in the typesafe variant so 
 the type can transfer -- composibility. The type-unsafe variant does look off. Let me look again to see what changed.

Reply:
I didn't change the type-unsafe one. It was already wierd, multiple-statements w/o  do/while:
This isn't generally anyone's preference but I left it alone. It is  used within a larger do/while and  is fairly private so ok.

```#define CLEAR_ICALL_FRAME_VALUE(RESULT, HANDLE)				\
	mono_stack_mark_record_size (__info, &__mark, __FUNCTION__);	\
	(RESULT) = mono_stack_mark_pop_value (__info, &__mark, (HANDLE));
```

Reply:
Given the type parameter, it might be possible to keep it a statement. Though that would not seem preferable.

Reply:
In one case `CLEAR_ICALL_FRAME_VALUE` sets the `RESULT` value and in the other it returns it, this is using the same symbol to do 2 different things, which makes it harder to read. I understand in one case - the type safe case - we need to pass a type to actually be type safe, but changing from statement to expression is purely because of your preference for the latter. Nothing forces you to use a expression over a statement, and you could simply - as you also pointed out - use a statement with an added parameter. I will not block this PR over that but stop confusing your preferences over required differences, it's not the first time and it's becoming wearing.